File: 1701-1800/1748-sum-of-unique-elements.rs
struct Solution {}

impl Solution {
    pub fn sum_of_unique(nums: Vec<i32>) -> i32 {
        use std::collections::HashMap;
        let mut cnts = HashMap::new();
        for x in nums {
            *cnts.entry(x).or_insert(0) += 1;
        }
        let mut ans = 0;
        for (k, v) in cnts {
            if v == 1 {
                ans += k;
            }
        }
        ans
    }
}

fn main() {
    assert_eq!(Solution::sum_of_unique(vec![1,2,3,2]), 4);
}

File: 1701-1800/1748-sum-of-unique-elements.cpp
class Solution {
public:
    int sumOfUnique(vector<int>& nums) {
        array<int, 101> cnts{0};
        for (int x : nums) ++cnts[x];
        int ans = 0;
        for (int i = 1; i <= 100; ++i) {
            if (cnts[i] == 1)
                ans += i;
        }
        return ans;
    }
};
